src/repositories/jobs.js

```diff
@@ -14,9 +15,63 @@ const create = async (payload) => {
   }
 };
 
+const createWithRecruiter = async (payload) => {
+  try {
+    let userId = payload.user_uuid;
+    let recruiterName = payload.recruiter_name;
+    let recruiterDescription = payload.recruiter_description;
+    delete payload.recruiter_description;
+    delete payload.recruiter_name
+    delete payload.user_uuid
+    let recruiter = await knex('recruiters').where({ user_uuid: userId }).first();
+    console.log(recruiter,"existing-rec")
+    if (recruiter) {
+      payload = baseRepo.addCreatedTimestamps(payload);
+      payload['recruiter_uuid'] = recruiter.uuid;
+      console.log(payload,"job-payload")
+      let result = await knex.transaction(async (trx) => {
+        const rows = await trx("jobs").insert(payload).returning("*");
+        return rows[0];
+      });
+      return {
+        recruiter:recruiter,
+        job:result
+      };
+    } else {
+      console.log("else-else")
+      let recruiterPayload = {}
+      recruiterPayload['name'] = recruiterName;
+      recruiterPayload['recruiter_type'] = "in-house";
```

Comment:
in-house should come from the payload and not hardcoded in this function. It won't scale.